CDP4Composition/Services/PluginUpdaterService/PluginFileSystemService.cs

```diff
@@ -0,0 +1,135 @@
+//    compliant with ECSS-E-TM-10-25 Annex A and Annex C.
+//
+//    The CDP4-IME Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4Composition.Services.PluginUpdaterService
+{
+    using System.ComponentModel.Composition;
+    using System.IO;
+    using System.IO.Compression;
+
+    using CDP4Composition.Modularity;
+
+    using NLog;
+
+    /// <summary>
+    /// The purpose of the <see cref="PluginFileSystemService"/> is to provide operations appliable on a file system
+    /// </summary>
+    [Export(typeof(IPluginFileSystemService))]
+    public class PluginFileSystemService : IPluginFileSystemService
+    {
+        /// <summary>
+        /// Gets or sets the path were the cdp4ck to be installed sits 
+        /// </summary>
+        public FileInfo UpdateCdp4CkFileInfo { get; set; }
+
+        /// <summary>
+        /// Gets or sets the path where the old version of the plugin will be temporaly kept in case anything goes wrong with the installation
+        /// </summary>
+        public DirectoryInfo TemporaryPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the path where the updated plugin should be installed
+        /// </summary>
+        public DirectoryInfo InstallationPath { get; set; }
+
+        /// <summary>
+        /// Instanciate a new <see cref="PluginFileSystemService"/>
+        /// </summary>
+        /// <param name="plugin"></param>
+        public PluginFileSystemService((FileInfo cdp4ckFile, Manifest manifest) plugin)
+        {
+            var (cdp4CkFile, manifest) = plugin;
+
+            this.UpdateCdp4CkFileInfo = cdp4CkFile;
+            this.InstallationPath = PluginUtilities.GetPluginDirectory(manifest.Name);
+            this.TemporaryPath = PluginUtilities.GetTempDirectoryInfo(manifest.Name);
+        }
+
+        /// <summary>
+        /// Installs the new version into the <see cref="InstallationPath"/>
+        /// </summary>
+        public void InstallNewVersion()
+        {
+            if (!this.InstallationPath.Exists)
+            {
+                this.InstallationPath.Create();
+            }
+
+            using var zip = ZipFile.OpenRead(this.UpdateCdp4CkFileInfo.FullName);
```

Comment:
shouldnt the `using` encapsulate the below code?

Reply:
the ZipArchive will get disposed anyway as soon as the execution leaves the block of code with or without the extra braquets, and It doesnt need to be disposed sooner. This syntax is lighter. But if I'm making a styling mistake I'll stop using using declaration

Reply:
please do, we should put explicit code blocks, for the same reason we do not do one-line or two-line `if` statements.

Reply:
It makes sense